htlcswitch/quiescer.go

```diff
@@ -140,6 +160,10 @@ type QuiescerLive struct {
 	// channel was quiescent.
 	resumeQueue []func()
 
+	// timeoutTimer is a field that is used to hold onto the timeout job
+	// when we reach quiescence.
+	timeoutTimer fn.Option[*time.Timer]
```

Comment:
why is this an option? shouldn't we always have this timeout?

Reply:
What I meant was this didn't feel like an option, it's technically wrong when we want a timer there but it's `fn.None`. On the other hand, when we start a timeout we don't need to worry about canceling the previous one - if there's a previous one, think it's also an invalid state. Easier to demonstrate in a diff,

```diff
diff --git a/htlcswitch/quiescer.go b/htlcswitch/quiescer.go
index f23e183d1..5995e24fb 100644
--- a/htlcswitch/quiescer.go
+++ b/htlcswitch/quiescer.go
@@ -167,7 +167,7 @@ type QuiescerLive struct {
 
 	// timeoutTimer is a field that is used to hold onto the timeout job
 	// when we reach quiescence.
-	timeoutTimer fn.Option[*time.Timer]
+	timeoutTimer *time.Timer
 
 	sync.RWMutex
 }
@@ -540,10 +540,7 @@ func (q *QuiescerLive) startTimeout() {
 		return
 	}
 
-	q.timeoutTimer.WhenSome(func(t *time.Timer) { t.Stop() })
-	q.timeoutTimer = fn.Some(
-		time.AfterFunc(q.cfg.timeoutDuration, q.cfg.onTimeout),
-	)
+	q.timeoutTimer = time.AfterFunc(q.cfg.timeoutDuration, q.cfg.onTimeout)
 }
 
 // cancelTimeout cancels the timeout function that would otherwise fire if the
@@ -551,8 +548,7 @@ func (q *QuiescerLive) startTimeout() {
 // before startTimeout or after another call to cancelTimeout, the effect will
 // be a noop.
 func (q *QuiescerLive) cancelTimeout() {
-	q.timeoutTimer.WhenSome(func(t *time.Timer) { t.Stop() })
-	q.timeoutTimer = fn.None[*time.Timer]()
+	q.timeoutTimer.Stop()
 }
 
 type quiescerNoop struct{}
```

Reply:
I can remove the option but it does mean that we end up with a nil timer during initialization but then we need to do other things to defend against when it is nil. As it is right now the API has safe semantics around out-of-order calls and back-to-back calls.